internal/provider/resource_ssh_key.go

```diff
@@ -0,0 +1,257 @@
+			Name:        oxide.Name(plan.Name.ValueString()),
+			PublicKey:   plan.PublicKey.ValueString(),
+		},
+	}
+
+	sshKey, err := r.client.CurrentUserSshKeyCreate(params)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating SSH key",
+			"API error: "+err.Error(),
+		)
+		return
+	}
+	tflog.Trace(ctx, fmt.Sprintf("created SSH key with ID: %v", sshKey.Id), map[string]any{"success": true})
+
+	// Map response body to schema and populate computed attribute values.
+	plan.ID = types.StringValue(sshKey.Id)
+	plan.SiloUserID = types.StringValue(sshKey.SiloUserId)
+	plan.TimeCreated = types.StringValue(sshKey.TimeCreated.String())
+	plan.TimeModified = types.StringValue(sshKey.TimeModified.String())
+
+	// Save plan into Terraform state.
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *sshKeyResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state sshKeyResourceModel
+
+	// Read Terraform prior state data into the model.
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	readTimeout, diags := state.Timeouts.Read(ctx, defaultTimeout())
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	ctx, cancel := context.WithTimeout(ctx, readTimeout)
+	defer cancel()
+
+	sshKey, err := r.client.CurrentUserSshKeyView(oxide.CurrentUserSshKeyViewParams{
+		SshKey: oxide.NameOrId(state.ID.ValueString()),
+	})
+	if err != nil {
+		if is404(err) {
+			// Remove resource from state during a refresh
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError(
+			"Unable to read SSH key:",
+			"API error: "+err.Error(),
+		)
+		return
+	}
+	tflog.Trace(ctx, fmt.Sprintf("read SSH key with ID: %v", sshKey.Id), map[string]any{"success": true})
+
+	state.Description = types.StringValue(sshKey.Description)
+	state.ID = types.StringValue(sshKey.Id)
+	state.Name = types.StringValue(string(sshKey.Name))
+	state.PublicKey = types.StringValue(string(sshKey.PublicKey))
+	state.SiloUserID = types.StringValue(string(sshKey.SiloUserId))
+	state.TimeCreated = types.StringValue(sshKey.TimeCreated.String())
+	state.TimeModified = types.StringValue(sshKey.TimeModified.String())
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update is intentionally unimplemented since SSH keys do not have an update
+// API. All of its configurable attributes are marked as requiring replacement
+// to tell Terraform to destroy and create this resource upon change to its
+// attributes. If an update API is created in the future this method should be
+// implemented.
+func (r *sshKeyResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+}
```

Comment:
Let's add an error here just for completeness.

```suggestion
	resp.Diagnostics.AddError(
		"Error updating SSH key",
		"the oxide API currently does not support updating SSH keys")
}
```